rust/private/rustc.bzl

```diff
@@ -167,24 +169,29 @@ def collect_deps(label, deps, proc_macro_deps, aliases, toolchain):
             transitive_crates.append(depset([dep[rust_common.crate_info]], transitive = [dep[DepInfo].transitive_crates]))
             transitive_dylibs.append(dep[DepInfo].transitive_dylibs)
             transitive_staticlibs.append(dep[DepInfo].transitive_staticlibs)
+            transitive_staticlibs_alwayslink.append(dep[DepInfo].transitive_staticlibs_alwayslink)
             transitive_build_infos.append(dep[DepInfo].transitive_build_infos)
         elif CcInfo in dep:
             # This dependency is a cc_library
 
             # TODO: We could let the user choose how to link, instead of always preferring to link static libraries.
-            libs = get_libs_for_static_executable(dep)
-
-            transitive_dylibs.append(depset([
-                lib
-                for lib in libs.to_list()
-                # Dynamic libraries may have a version number nowhere, or before (macos) or after (linux) the extension.
-                if lib.basename.endswith(toolchain.dylib_ext) or lib.basename.split(".", 2)[1] == toolchain.dylib_ext[1:]
-            ]))
-            transitive_staticlibs.append(depset([
-                lib
-                for lib in libs.to_list()
-                if lib.basename.endswith(toolchain.staticlib_ext)
-            ]))
+            linker_inputs = dep[CcInfo].linking_context.linker_inputs.to_list()
+            for li in linker_inputs:
+                always = []
+                static = []
+                dylib = []
+                for lib in li.libraries:
+                    art = get_preferred_artifact(lib)
+                    if lib.alwayslink:
+                        always.append(lib.static_library or lib.pic_static_library)
+                        # Dynamic libraries may have a version number nowhere, or before (macos) or after (linux) the extension.
```

Comment:
dedent the comment to match the line it relates to

Reply:
This is now (mercifully) gone.